rasa_nlu/classifiers/embedding_intent_classifier.py

```diff
@@ -0,0 +1,565 @@
+                len(self.hidden_layer_size_b) != self.num_hidden_layers_b):
+            if len(self.hidden_layer_size_b) == 0:
+                raise ValueError("hidden_layer_size_b = {} "
+                                 "is an empty list, "
+                                 "while num_hidden_layers_b = {} > 0"
+                                 "".format(self.hidden_layer_size_b,
+                                           self.num_hidden_layers_b))
+
+            logger.error("The length of hidden_layer_size_b  = {} "
+                         "does not correspond to num_hidden_layers_b "
+                         "= {}. Set hidden_layer_size_b to "
+                         "the first element = {} for all layers"
+                         "".format(len(self.hidden_layer_size_b),
+                                   self.num_hidden_layers_b,
+                                   self.hidden_layer_size_b[0]))
+
+            self.hidden_layer_size_b = self.hidden_layer_size_b[0]
+
+        if not isinstance(self.hidden_layer_size_b, list):
+            self.hidden_layer_size_b = [self.hidden_layer_size_b
+                                        for _ in range(
+                                            self.num_hidden_layers_b)]
+
+        self.batch_size = self.component_config['batch_size']
+        self.epochs = self.component_config['epochs']
+
+        # embedding parameters
+        self.embed_dim = self.component_config['embed_dim']
+        self.mu_pos = self.component_config['mu_pos']
+        self.mu_neg = self.component_config['mu_neg']
+        self.similarity_type = self.component_config['similarity_type']
+        self.num_neg = self.component_config['num_neg']
+        self.use_max_sim_neg = self.component_config['use_max_sim_neg']
+
+        # regularization
+        self.C2 = self.component_config['C2']  # l2
+        self.C_emb = self.component_config['C_emb']
+        self.droprate = self.component_config['droprate']  # dropout
+
+        # flag if tokenize intents
+        self.intent_tokenization_flag = self.component_config[
+                                            'intent_tokenization_flag']
+        self.intent_split_symbol = self.component_config[
+                                        'intent_split_symbol']
+
+        # transform intents to numbers
+        # encode intents with numbers
+        self.intent_dict = intent_dict
+        # encode words in intents with numbers
+        self.intent_token_dict = intent_token_dict
+
+        # tf related instances
+        self.session = session
+        self.graph = graph if graph is not None else tf.Graph()
+        self.embedding_placeholder = embedding_placeholder
+        self.intent_placeholder = intent_placeholder
+        self.similarity_op = similarity_op
+
+    @classmethod
+    def required_packages(cls):
+        # type: () -> List[Text]
+        return ["tensorflow"]
+
+    def train(self, training_data, cfg=None, **kwargs):
```

Comment:
again, a lot of the stuff in this function can go in separate functions to be called from here. As you can probably see in code climate, the maximum function length is technically 25

Reply:
do you think it makes sense to add additional helper functions in load and persist?